lucet-runtime/lucet-runtime-internals/src/instance.rs

```diff
@@ -856,30 +860,41 @@ impl Instance {
             &args_with_vmctx,
         )?;
 
-        // Set up the guest to set itself as terminable, then continue to
-        // whatever guest code we want to run.
-        //
-        // `lucet_context_activate` takes two arguments:
-        // rsi: address of guest code to execute
-        // rdi: pointer to a bool that indicates the guest can be terminated
-        //
-        // The appropriate value for `rsi` is the top of the guest stack, which
-        // we would otherwise return to and start executing immediately. For
-        // `rdi`, we want to pass a pointer to the instance's `terminable` flag.
-        //
-        // once we've set up arguments, swap out the guest return address with
-        // `lucet_context_activate` so we start execution there.
+        self.with_terminability().swap_and_return()
+    }
+
+    /// Prepare the guest so that it will mark itself as terminable upon entry.
+    ///
+    /// This mutates the context's registers so that an activation function that will be run after
+    /// performing a context switch. This function (`enter_guest_region`) will mark the guest as
+    /// terminable before continuing to whatever guest code we want to run.
+    ///
+    /// `lucet_context_activate` takes three arguments in the following registers:
+    ///   * rdi: the data for the entry callback.
+    ///   * rsi: the address of the entry callback.
+    ///   * rbx: the address of the guest code to execute.
+    ///
+    /// The appropriate value for `rbx` is the top of the guest stack, which we would otherwise
+    /// return to and start executing immediately. For `rdi`, we want to pass our callback data
+    /// (a raw pointer to the instance). This will be passed as the first argument to the entry
+    /// function, which is responsible for updating the kill state's execution domain.
+    ///
+    /// See `lucet_runtime_internals::context::lucet_context_activate`, and
+    /// `execution::enter_guest_region` for more info.
+    fn with_terminability(&mut self) -> &mut Self {
```

Comment:
:hot_pepper: So, _this_ is a spicy change, but it's a no-op in the name of documentation. :hot_pepper:

We have our activation thunk, our entry callback, and data to pass to the entry callback. Notably, the activation thunk does not _actually_ care what the entry callback does, or what data is passed to it! The activation thunk exists purely to call some callback, and then jump to an address.

_This_ block, which was previously inlined in `run_func`, is how that thunk is used to make the guest terminable upon entry. Putting this into a separate private method means we can document this a little more cleanly.

I am open to bikeshedding name here if you'd like, I know returning `&mut Self` means this isn't quite what we would canonically call `with_something`. I just liked that this means we can place this into `run_func` above inside of a line that looks like:

```rust
self.with_terminability().swap_and_return()
```